photon-api/src/main/scala/com/linkedin/photon/ml/evaluation/PrecisionAtKMultiEvaluator.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.evaluation
+
+import org.apache.spark.rdd.RDD
+
+import com.linkedin.photon.ml.Types._
+
+/**
+ * Batch version of the precision@k evaluator.
+ *
+ * @param k The cut-off rank based on which the precision is computed
+ * @param idTag Name of the column or metadata field containing IDs used to group samples (e.g. documentId or queryId)
+ * @param ids A [[RDD]] of (unique sample identifier, ID) pairs. The IDs are used to group samples, then the evaluation
+ *            metric is computed on the groups per-ID and averaged. Such IDs can be thought of as a recommendation
+ *            context (e.g. queryId when evaluating the relevance of search results for given queries).
+ * @param labelAndOffsetAndWeights A [[RDD]] of (unique sample identifier, (label, offset, weight)) pairs
+ */
+protected[ml] class PrecisionAtKMultiEvaluator(
+    k: Int,
+    idTag: String,
+    override protected[ml] val ids: RDD[(UniqueSampleId, String)],
```

Comment:
This one actually needs it: it's marking that this `ids` is one-and-the-same with the `ids` declared in the constructor of the base class.